packages/react-widgets/src/models/utils.js

```diff
@@ -0,0 +1,42 @@
+import { AggregationTypes, _filtersToSQL } from '@carto/react-core';
+
+export function formatTableNameWithFilters(props) {
+  const { source, spatialFilter } = props;
+  const { data, filters, filtersLogicalOperator } = source;
+
+  const filtersClause = _filtersToSQL(filters, filtersLogicalOperator);
+
+  let query = `(${data} ${filtersClause}) foo`;
+
+  if (spatialFilter) {
+    const spatialFilterString = JSON.stringify(spatialFilter);
+    const spatialFilterClause = `ST_Intersects(geom, ST_GEOGFROMGEOJSON("""${spatialFilterString}"""))`;
+    query += ` WHERE ${spatialFilterClause}`;
+  }
+
+  return query;
+}
+
+// Operation columns & Join operation
+const SELECT_CLAUSE_BY_JOIN_OPERATION = {
+  [AggregationTypes.AVG]: (operationColumns) =>
+    `(${operationColumns.join(' + ')} / ${operationColumns.length})`,
+  [AggregationTypes.SUM]: (operationColumns) => `(${operationColumns.join(' + ')})`,
+  [AggregationTypes.COUNT]: (operationColumns) => operationColumns[0],
+  [AggregationTypes.MIN]: (operationColumns) => `(${operationColumns.join(' + ')})`,
+  [AggregationTypes.MAX]: (operationColumns) => `(${operationColumns.join(' + ')})`
```

Comment:
I'm trying to think about 'examples' for a FormulaWidget using several columns and min or max... to understand better usecase, do you have any @Clebal ?

Reply:
Hmmm I don't have any real example tbh. I think that multiple columns doesn't make sense with some AggregationTypes like count or has limited use cases like in min/max.

I guess that I would want to have multiple columns in FormulaWidget to show the avg of min/max months of the stores.

```
<FormulaWidget
    title="Average revenue in the worst month"
    column={["revenue_jan", "revenue_feb", ...]}
    joinOperation={AggregationTypes.MIN}
    operation={AggregationTypes.AVG}
/>
```

Reply:
Thx for the example... But if that's the case, isn't the SELECT wrong using ' + ' that way @Clebal ?

Reply:
Yes, this is already changed.

New code is:
```
[AggregationTypes.MIN]: (operationColumns) => `LEAST(${operationColumns.join()})`,
[AggregationTypes.MAX]: (operationColumns) => `GREATEST(${operationColumns.join()})`
```